src/main/java/liquibase/nosql/database/AbstractNoSqlDatabase.java

```diff
@@ -0,0 +1,329 @@
+package liquibase.nosql.database;
+
+/*-
+ * #%L
+ * Liquibase NoSql Extension
+ * %%
+ * Copyright (C) 2020 Mastercard
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+
+import liquibase.CatalogAndSchema;
+import liquibase.database.AbstractJdbcDatabase;
+import liquibase.database.Database;
+import liquibase.database.DatabaseConnection;
+import liquibase.exception.DatabaseException;
+import liquibase.exception.LiquibaseException;
+import liquibase.exception.ValidationErrors;
+import liquibase.statement.DatabaseFunction;
+import liquibase.structure.DatabaseObject;
+import lombok.NoArgsConstructor;
+
+import java.math.BigInteger;
+import java.util.Collections;
+import java.util.List;
+
+import static java.util.Optional.ofNullable;
+
+/**
+ * {@link AbstractNoSqlDatabase} is extended by all supported NoSql databases as a facade to the underlying database.
+ * The physical connection can be retrieved from the {@link AbstractNoSqlDatabase} implementation, as well as any
+ * database-specific characteristics.
+ */
+@NoArgsConstructor
+public abstract class AbstractNoSqlDatabase extends AbstractJdbcDatabase implements Database {
+
+    @Override
+    public int getPriority() {
+        return PRIORITY_DATABASE;
+    }
+
+    @Override
+    public boolean supportsInitiallyDeferrableColumns() {
+        return false;
+    }
+
+    @Override
+    public boolean supportsSequences() {
+        return false;
+    }
+
+    @Override
+    public boolean supportsDropTableCascadeConstraints() {
+        return false;
+    }
+
+    @Override
+    public boolean supportsAutoIncrement() {
+        return false;
+    }
+
+    @Override
+    public String getLineComment() {
+        return "//";
```

Comment:
@filipelautert: what's this supposed to do / does this need to be implemented? we're not generating SQL but JSON (though we're not even generating that, we're taking whatever the changeset contains) and that doesn't support comments

Reply:
This is used to generateSQL and a comment is loaded from database metadata. I think you can ignore by now until a snapshot is being implemented . But if opensearch doesn't support comment metadata then it will never be called.

Reply:
it doesn't support comments and snapshots wouldn't be generated as SQL. so i'm wondering if this shouldn't instead throw an exception?

Reply:
done

Reply:
turns out that was a bad idea - this now crashes everything (and we didn't catch it because the CI currently doesn't work 😢). `ChangeLogParameters` always calls `AbstractNoSqlDatabase#getLineComment`, so i'll just return an empty string instead